.github/scripts/check_last_update.py

```diff
@@ -0,0 +1,96 @@
+#! /usr/bin/env python3
+
+from collections import defaultdict
+from pathlib import Path
+import argparse
+import json
+import os
+import re
+import sys
+
+
+def extractUpdateDate(content):
+    """Extract update date"""
+
+    update_date = None
+    date_pattern = re.compile(
+        '.*{:\s?datetime="(?P<update>[0-9]{4}-[0-9]{2}-[0-9]{2})"\s?}.*'
+    )
+    for line in content:
+        matches = date_pattern.match(line)
+        if matches:
+            dates = matches.groupdict()
+            return dates["update"]
+
+    return update_date
+
+
+def findAllFiles(path):
+    """Create a list of all markdown files in path"""
+
+    files = defaultdict(list)
+    search_path = Path(path)
+    file_paths = search_path.glob(f"*/*.md")
+
+    for fp in file_paths:
+        # Threat the first folder as locale code
+        locale = str(fp).split(os.sep)[0]
+        filename = os.path.relpath(fp, os.path.join(path, locale))
+        files[locale].append(filename)
+
+    return files
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--path",
+        required=True,
+        dest="files_path",
+        help="Path to folder including subfolders for all locales",
+    )
+    parser.add_argument(
+        "--ref",
+        required=False,
+        dest="ref_locale",
+        default="en",
+        help="Reference locale code (default 'en')",
+    )
+    args = parser.parse_args()
+
+    files_list = findAllFiles(args.files_path)
+
+    data = defaultdict(dict)
+    for locale, filenames in files_list.items():
+        for f in filenames:
+            filename = os.path.join(args.files_path, locale, f)
+            with open(filename, "r") as fp:
+                content = fp.readlines()
```

Comment:
I think it'd be more efficient to just pass `fp` to `extractUpdateDate` and let that function use `for line in fp` since that will only read the lines you need rather than reading them all into memory. Not a big deal but might be slightly faster and a bit more memory efficient.

Reply:
Yes, definitely more efficient, considering the date is always at the beginning. Ended up passing the filename, and delegating the whole file management to the function.